src/main/java/com/monitorjbl/xlsx/sst/FileBackedList.java

```diff
@@ -49,9 +52,10 @@ public boolean removeEldestEntry(Map.Entry eldest) {
 
   public void add(String str) {
     try {
-      if (str!=null && str.length()>0) {
+      if (cacheSize > 0 || (str!=null && str.length()>0)) {
```

Comment:
Are you sure that empty (but not null) strings shouldn't be added to the shared string table?
Not including these will put the other referenced strings out of alignments.